pkg/compare/parsing.go

```diff
@@ -142,7 +142,7 @@ func parseTemplates(templatePaths, functionTemplates []string, fsys fs.FS) ([]*t
 			continue
 		}
 		// recreate template with new name that includes path from reference root:
-		parsedTemp, _ = template.New(temp).Funcs(FuncMap()).AddParseTree(temp, parsedTemp.Tree)
+		parsedTemp, _ = parsedTemp.New(temp).Funcs(FuncMap()).AddParseTree(temp, parsedTemp.Lookup(path.Base(temp)).Tree)
```

Comment:
Curious why we are recreating template? 

If we init just one time (L139) then we can simply do `parsedTemp, _ = parsedTemp.Funcs(FuncMap()).AddParseTree(temp, parsedTemp.Tree)` ?

Reply:
Also I think if we don't recreate the template I don't think we need to re-add the the funcmap

Reply:
we recreate the template because we need to change its name, for future usage in the code.
the name gets automatically choose as the name of the file, that is not good for us in case of templates included in sub-directories, there is now way to change only the name, then we recreate it with the name we want including the relative path from the metadata directory 


maybe after we merge some of @nocturnalastro PRs we can add a name attribute to the wrapped template instead, need to still think about it. but that is another story.